app/src/main/java/com/github/se/cyrcle/ui/review/AllReviewsScreen.kt

```diff
@@ -57,27 +61,30 @@ fun AllReviewsScreen(
     reviewViewModel: ReviewViewModel,
     userViewModel: UserViewModel
 ) {
-  val selectedParking =
-      parkingViewModel.selectedParking.collectAsState().value
-          ?: return Text(text = "No parking selected. Should not happen")
 
-  val (selectedCardIndex, setSelectedCardIndex) = remember { mutableStateOf(-1) }
+  val selectedParking by parkingViewModel.selectedParking.collectAsState()
+  val reviewsList by reviewViewModel.parkingReviews.collectAsState()
+  val ownerHasReviewed = remember { mutableStateOf(false) }
+  val (selectedCardIndex, setSelectedCardIndex) = remember { mutableIntStateOf(-1) }
 
-  val ownerHasReviewed =
-      if (userViewModel.currentUser.value?.public?.userId != null) {
-        reviewViewModel.parkingReviews.value.any {
-          it.owner == userViewModel.currentUser.value?.public?.userId
-        }
-      } else {
-        false
-      }
+  // When the screen is composed, clear the reviews and fetch the reviews for the selected parking
+  LaunchedEffect(Unit) {
+    reviewViewModel.clearReviews()
```

Comment:
why do the reviews need to be cleared first?

Reply:
In the scenario where a user first look at the review of spot1 and then look at the reviews of spot2.
The user would still see the reviews of spot1 while the reviews of spot2 are fetched from the db

Reply:
Thanks for the example!